src/uproot/_dask.py

```diff
@@ -431,6 +432,21 @@ def real_filter_branch(branch):
 
     dask_dict = {}
 
+    start_stop_step_map = {}
+    for i, ttree in enumerate(hasbranches):
+        entry_start, entry_stop = _regularize_entries_start_stop(
+            ttree.num_entries, None, None
+        )
+
+        branchid_interpretation = {}
+        for key in common_keys:
+            branch = ttree[key]
+            branchid_interpretation[branch.cache_key] = branch.interpretation
+        entry_step = _regularize_step_size(
+            ttree, step_size, entry_start, entry_stop, branchid_interpretation
+        )
+        start_stop_step_map[i] = (entry_start, entry_stop, entry_step)
```

Comment:
If we have a `hasbranches` list and we want to map a tree to certain properties (like an entry start,stop and step here) what would be the most appropriate way to hash the ttree for the map? I have used the order in which `enumerate(ttree)` loops over them, but I'm not sure if that's the best practise. Please suggest.

Reply:
cc @jpivarski @agoose77

Reply:
The `entry_start`, `entry_stop` data does not need to come from

```python
        entry_start, entry_stop = _regularize_entries_start_stop(
            ttree.num_entries, None, None
        )
```

Since we're doing whole TTrees, `entry_start = 0` and `entry_stop = ttree.num_entries`. A TTree's `num_entries` is already a lookup; creating a new lookup table for it won't optimize anything. (Though removing the `_regularize_entries_start_stop` could save a bit of time.)

It's also not necessary to have fine-tuned `step_size` per file. They differ by file because each file has slightly different numbers of items per entry, but a set of ROOT files almost always represents an [independent and identically distributed](https://en.wikipedia.org/wiki/Independent_and_identically_distributed_random_variables) dataset, so the `step_size` of each file will be roughly equal. Also, chunk sizes don't need to be exact, since their purpose is to protect from memory issues on one side and overhead on the other, which is an order-of-magnitude thing.

So you can average the `step_size` values of all the files, round it to the nearest integer and ensure that it's at least 1. (Probably not necessary because the `step_size` of each file is already guaranteed to be at least 1, so their average will be as well.) The `step_size` for a single file is already an average over TBaskets.